metadata/src/main/java/org/apache/kafka/image/loader/MetadataBatchLoader.java

```diff
@@ -100,7 +100,7 @@ public boolean hasSeenRecord() {
      */
     public final void resetToImage(MetadataImage image) {
         this.image = image;
-        this.hasSeenRecord = true;
+        this.hasSeenRecord = !image.isEmpty();
```

Comment:
This was an existing issue that somehow became easier to trigger due to these changes. Thanks to @mumrah for figuring out that this was the reason for the remaining failures.